.travis.yml

```diff
@@ -9,57 +11,41 @@ jobs:
     env:
       - CAN_FAIL=true
   include:
+    - stage: prepare cache
+      name: "Prepare the Ruby cache"
+      script: true
+      before_script:
+        - gem install asciidoctor asciidoctor-diagram rouge
+
     # fail if asciidoctor encounters errors
     - stage: validate
       name: "Validate updated assemblies with Asciidoctor"
-      install:
-        - gem install asciidoctor
-        - gem install asciidoctor-diagram
-        - gem install rouge
+      language: ruby
       script:
         - chmod +x ./scripts/check-asciidoctor-build.sh
         - ./scripts/check-asciidoctor-build.sh
+
     - stage: build
+      language: python
       name: "Build openshift-enterprise distro"
-      before_install:
-        - gem install asciidoctor
-        - gem install asciidoctor-diagram
-      install:
-        - pip3 install pyyaml
-        - pip3 install aura.tar.gz
       script:
         - python3 build.py --distro openshift-enterprise --product "OpenShift Container Platform" --version 4.14 --no-upstream-fetch && python3 makeBuild.py
     - # stage name not required, will continue to use `build`
       if: branch IN (main, enterprise-4.13, enterprise-4.14)
       name: "Build openshift-dedicated distro"
-      before_install:
-        - gem install asciidoctor
-        - gem install asciidoctor-diagram
-      install:
-        - pip3 install pyyaml
-        - pip3 install aura.tar.gz
+      language: python
       script:
         - python3 build.py --distro openshift-dedicated --product "OpenShift Dedicated" --version 4 --no-upstream-fetch && python3 makeBuild.py
     - # stage name not required, will continue to use `build`
       if: branch IN (main, enterprise-4.13, enterprise-4.14)
       name: "Build openshift-rosa distro"
-      before_install:
-        - gem install asciidoctor
-        - gem install asciidoctor-diagram
-      install:
-        - pip3 install pyyaml
-        - pip3 install aura.tar.gz
+      language: python
       script:
         - python3 build.py --distro openshift-rosa --product "Red Hat OpenShift Service on AWS" --version 4 --no-upstream-fetch && python3 makeBuild.py
     - # stage name not required, will continue to use `build`
       if: branch IN (main, enterprise-4.13, enterprise-4.14)
       name: "Build microshift distro"
-      before_install:
-        - gem install asciidoctor
-        - gem install asciidoctor-diagram
-      install:
-        - pip3 install pyyaml
-        - pip3 install aura.tar.gz
+      language: python
```

Comment:
**issue:** You must also update the sequence of stages L77. Without it the Cache stage runs at the end. 
![image](https://github.com/openshift/openshift-docs/assets/23069445/403fa886-6615-48f9-9550-3d7c9813082a)